test/src/test/java/org/corfudb/infrastructure/log/StreamLogFilesTest.java

```diff
@@ -1,121 +1,160 @@
         return new ServerContextBuilder()
-            .setLogPath(path)
-            .setMemory(false)
-            .build();
+                .setLogPath(path)
+                .setMemory(false)
+                .build();
     }
-    
+
     @Test
     public void testWriteReadWithChecksum() {
         // Enable checksum, then append and read the same entry
         StreamLog log = new StreamLogFiles(getContext(), false);
+        try {
+            ByteBuf b = Unpooled.buffer();
+            byte[] streamEntry = "Payload".getBytes();
+            Serializers.CORFU.serialize(streamEntry, b);
+            long address0 = 0;
+            log.append(address0, new LogData(DataType.DATA, b));
+            assertThat(log.read(address0).getPayload(null)).isEqualTo(streamEntry);
+
+            // Disable checksum, then append and read then same entry
+            // An overwrite exception should occur, since we are writing the
+            // same entry.
+            final StreamLog newLog = new StreamLogFiles(getContext(), true);
+            assertThatThrownBy(() -> newLog.append(address0, new LogData(DataType.DATA, b)))
+                    .isInstanceOf(OverwriteException.class);
+            newLog.close();
+
+            assertThat(log.read(address0).getPayload(null)).isEqualTo(streamEntry);
+        } finally {
+            log.close();
+        }
+    }
+
+    @Test
+    public void testFileDescriptorsLeak() throws Exception {
+        StreamLogFiles log = new StreamLogFiles(getContext(), false);
         ByteBuf b = Unpooled.buffer();
         byte[] streamEntry = "Payload".getBytes();
         Serializers.CORFU.serialize(streamEntry, b);
         long address0 = 0;
         log.append(address0, new LogData(DataType.DATA, b));
-        assertThat(log.read(address0).getPayload(null)).isEqualTo(streamEntry);
 
-        // Disable checksum, then append and read then same entry
-        // An overwrite exception should occur, since we are writing the
-        // same entry.
-        final StreamLog newLog = new StreamLogFiles(getContext(), true);
-        assertThatThrownBy(() -> {
-            newLog.append(address0, new LogData(DataType.DATA, b));
-        })
-                .isInstanceOf(OverwriteException.class);
-        assertThat(log.read(address0).getPayload(null)).isEqualTo(streamEntry);
+        SegmentSupervisor segmentSupervisor = log.getSegmentSupervisor();
+        SegmentHandle segment = segmentSupervisor.getSegmentByAddress(0, sh -> {});
+        segmentSupervisor.close(segment);
+        log.contains(0);
+        // a resource leak could happen in case of clearing channels
+        // after the fact that `contains` method just added a segment into the Map.
+        // To prevent resources leak don't clean the map (SegmentSupervisor.channels).
+        //segmentSupervisor.getChannels().clear();
+        log.close();
+
+        segmentSupervisor.closeSegmentHandlers(0);
+        lsofSpec.check();
```

Comment:
Can't you execute this after every test with a junit annotation ?

Reply:
I do https://github.com/CorfuDB/CorfuDB/blob/e3d65380c753a1104f064f2e353b03cb55144882/test/src/test/java/org/corfudb/AbstractCorfuTest.java#L365

That's why it took so much time to bring this change. We have a big mess in our unit/integration tests and almost never release resources